csharp/src/Ice/PluginManagerI.cs

```diff
@@ -97,94 +85,54 @@ public void initializePlugins()
             _initialized = true;
         }
 
-        public string[] getPlugins()
+        public string[] GetPlugins()
         {
             lock (this)
             {
+                if (_state == StateDestroyed)
+                {
+                    throw new CommunicatorDestroyedException();
+                }
+
                 return _plugins.Select(p => p.Name).ToArray();
             }
         }
 
-        public Plugin getPlugin(string name)
+        public Plugin? GetPlugin(string name)
```

Comment:
I would simplify and merge GetPlugin and FindPlugin, and not check the state. It's no big deal to return a Plugin when the communicator is destroyed.

Reply:
FindPlugin is private and doesn't lock `_plugins` GetPluging does